source/server/init_manager_impl.cc

```diff
@@ -10,11 +10,24 @@
 namespace Envoy {
 namespace Server {
 
-InitManagerImpl::InitManagerImpl(absl::string_view description) : description_(description) {
+InitManagerImpl::InitManagerImpl(absl::string_view description)
+    : canceled_(std::make_shared<bool>(false)), description_(description) {
   TRACE_INIT_MANAGER("constructor");
 }
 
-InitManagerImpl::~InitManagerImpl() { TRACE_INIT_MANAGER("destructor"); }
+InitManagerImpl::~InitManagerImpl() {
+  switch (state_) {
+  case State::NotInitialized:
+    TRACE_INIT_MANAGER("destructor, not initialized");
+    break;
+  case State::Initializing:
+    TRACE_INIT_MANAGER("destructor, canceling initialization");
```

Comment:
nit: Can we just have 1 debug statement which prints the enum, and then an if statement for the case we really care about to set the cancellation flag.